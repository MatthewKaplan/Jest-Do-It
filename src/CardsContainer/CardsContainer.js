import React from 'react';
import '../Styles/_CardsContainer.scss';
import data from '../data.js';
import {Cards} from '../Cards/Cards.js';

export const CardsContainer = (props) => {
  const { currentCard } = props;
  let displayCard;

  data.jestQuestions.map( jest => 
    displayCard =
    <Cards 
      key={currentCard.id}
      id={jest.id}
      question={jest.question}
      potentialAnswers={jest.potentialAnswers}
      correctAnswer={jest.correctAnswer}
      onCorrectAnswer={jest.onCorrectAnswer}
      onIncorrectAnswer={jest.onIncorrectAnswer}
      linkName={jest.linkName}
      link={jest.link}
      currentCard={currentCard}
      checkAnswer={props.checkAnswer}
      answerResponse={props.answerResponse}
      nextCard={props.nextCard}
      activeButtons={props.activeButtons}
    />)
    
  return (
    <section className="CardsContainer">
      {displayCard}
    </section>
  );
}

export default CardsContainer;
