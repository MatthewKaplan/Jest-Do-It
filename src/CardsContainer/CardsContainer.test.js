import React from 'react';
import { CardsContainer } from './CardsContainer';
import { shallow } from 'enzyme';
