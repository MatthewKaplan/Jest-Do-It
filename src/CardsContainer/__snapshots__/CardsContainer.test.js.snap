// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer should match snapshot 1`] = `
<section
  className="CardsContainer"
>
  <Cards
    answerResponse=""
    checkAnswer={[Function]}
    correctAnswer=".toThrow()"
    currentCard={
      Object {
        "correctAnswer": "CORRECT ANSWER",
        "id": 23,
        "link": "https://jestjs.io/docs/en/tutorial-jquery#docsNav",
        "linkName": "Testing Events",
        "onCorrectAnswer": "Thats correct!",
        "onIncorrectAnswer": "Thats incorrect. The correct answer is: .simulate().",
        "potentialAnswers": Array [
          ".simulate()",
          ".onClick()",
          ".click()",
          "All of the above",
        ],
        "question": "How would one test a button or an input?",
        "save": "false",
      }
    }
    id={30}
    key="30"
    nextCard={[Function]}
    onCorrectAnswer="Thats correct!"
    onIncorrectAnswer="Thats incorrect. The correct answer is: .toThrow()."
    potentialAnswers={
      Array [
        ".toThrow()",
        ".toThrowError()",
        ".throwsError()",
        ".expectError()",
      ]
    }
    question="If you want to test that a particular function throws an error when it's called, use:"
  />
</section>
`;
